tests/cpp/test_sharding.cpp

```diff
@@ -23,26 +23,41 @@ namespace nvfuser {
 
 using ShardingTest = NVFuserFixtureParamTest<bool>;
 
-// TODO: This test checks that isSharded generates an error when a split/merged
-// axis is parallelized with DIDx. Update when this restriction is lifted.
-TEST_F(ShardingTest, IsSharded) {
+TEST_F(ShardingTest, LogicalIsSharded) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
-  TensorView* a = makeSymbolicTensor(3);
-  a->axis(2)->parallelize(ParallelType::DIDx);
-  a->split(0, 4);
-  EXPECT_TRUE(isSharded(a)) << "DIDx on logical domain";
+  TensorView* x = makeSymbolicTensor(3);
+  x->axis(2)->parallelize(ParallelType::DIDx);
+  x->split(0, 4);
 
-  TensorView* b = makeSymbolicTensor(3);
-  b->split(1, 4);
-  b->axis(1)->parallelize(ParallelType::DIDx);
-  EXPECT_TRUE(isSharded(b)) << "DIDx on loop domain";
```

Comment:
Due to the change in isSharded, this is changed to look at the allocation domain. Also, I split this test into three and expanded the error messages.